pkg/controller/operators/olm/overrides/inject_test.go

```diff
@@ -746,6 +746,30 @@ func TestInjectNodeSelectorIntoDeployment(t *testing.T) {
 				NodeSelector: map[string]string{"foo": "bar"},
 			},
 		},
+		{
+			// Existing PodSpec is left alone if nodeSelector is nil
+			// Expected: PodSpec is not changed
+			name: "WithNilNodeSelector",
+			podSpec: &corev1.PodSpec{
+				NodeSelector: defaultNodeSelector,
+			},
+			nodeSelector: nil,
+			expected: &corev1.PodSpec{
+				NodeSelector: defaultNodeSelector,
+			},
+		},
+		{
+			// Existing PodSpec is left alone if nodeSelector is an empty map
```

Comment:
This comment's out of sync with the test expectation. FWIW, I agree with the behavior as it is right now (nil doesn't override, empty does).

Reply:
Oops - addressed in the latest version of the PR.